frontend/packages/noobaa-storage-plugin/src/components/health-card/health-card.tsx

```diff
@@ -0,0 +1,145 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+
+import {
+  DashboardCard,
+  DashboardCardBody,
+  DashboardCardHeader,
+  DashboardCardTitle,
+} from '@console/internal/components/dashboard/dashboard-card';
+import { FirehoseResource } from '@console/internal/components/utils';
+import { HealthBody, HealthItem } from '@console/internal/components/dashboard/health-card';
+import { HealthState } from '@console/internal/components/dashboard/health-card/states';
+import {
+  withDashboardResources,
+  DashboardItemProps,
+} from '@console/internal/components/dashboards-page/with-dashboard-resources';
+import { referenceForModel, K8sResourceKind } from '@console/internal/module/k8s';
+import { NooBaaSystemModel } from '../../models';
+
+enum HealthCardQueries {
+  BUCKETS_COUNT = 'NooBaa_num_buckets',
+  UNHEALTHY_BUCKETS = 'NooBaa_num_unhealthy_buckets',
+  POOLS_COUNT = 'num_pools',
+  UNHEALTHY_POOLS = 'num_unhealthy_pools',
+}
+
+const noobaaSystemResource: FirehoseResource = {
+  kind: referenceForModel(NooBaaSystemModel),
+  namespaced: true,
+  namespace: 'openshift-storage',
+  name: 'noobaa-system',
+  isList: false,
+  prop: 'noobaa',
+};
+
+const getObjectStorageHealthState = (
+  bucketsResponse,
+  unhealthyBucketsResponse,
+  poolsResponse,
+  unhealthyPoolResponse,
+  noobaaSystemData,
+): ObjectStorageHealth => {
+  if (!(bucketsResponse && unhealthyBucketsResponse && poolsResponse && unhealthyPoolResponse)) {
+    return { state: HealthState.LOADING };
+  }
+  const buckets = _.get(bucketsResponse, 'data.result[0].value[1]');
```

Comment:
Buckets card already has this util function https://github.com/openshift/console/blob/master/frontend/packages/noobaa-storage-plugin/src/components/buckets-card/buckets-card.tsx#L26 which should be reused here too. Maybe move the function to some util file.